src/public/Interfaces.ts

```diff
@@ -75,16 +81,25 @@ export interface Context {
     res?: { [key: string]: any };
 }
 
-export interface HttpRequestHeaders {
-  [name: string]: string | undefined;
+/**
+ * HTTP request headers.
+ */
+export interface HttpHeaders {
+  [name: string]: string | unknown;
 }
 
+/**
+ * Query string parameter keys and values from the URL.
+ */
 export interface HttpRequestQuery {
-  [name: string]: string | undefined;
+  [name: string]: string | unknown;
```

Comment:
Same: `[name: string]: string`

Reply:
So the reason I made these `| undefined` is because otherwise you can access "any" property without confirming it actually exists - this is meant to gel with being able to use declaration merging to define properties that you know will exist.

i.e

```
import { HttpRequest } from '@azure/functions';

declare module '@azure/functions' {
  export interface HttpRequestHeaders {
    'x-github-event': GithubEventName;
    'x-github-delivery': string;
    'x-hub-signature': string;
    'x-hub-signature-256': string;
  }
}

export const getEvent = (request: HttpRequest): GithubEvent => {
  const rawBody = request.rawBody as string;
  const signature: string = request.headers['x-hub-signature-256'];
  const { GH_WEBHOOK_SECRET } = process.env;

  if (verify(GH_WEBHOOK_SECRET, rawBody, signature)) {
    // ensure we're referencing a header that is defined, outside of the cast
    const name = request.headers['x-github-event'];
    const payload = JSON.parse(rawBody) as GithubEvent['payload'];

    return { name, payload } as GithubEvent;
  }

  throw new Error('event did not come from github');
};
```

In the above, I don't have to do any checking on if the headers are `undefined` or not before using them as I've explicitly typed them as being just of type `string` (effectively I've told TS that they'll always be there).

Removing the `| undefined` from this signature will mean I could mistakenly typo a header and not be told - a great example of this in action was when I originally wrote this code: it was copied from an AWS Lambda, which preserves the headers original casing.

By having `| undefined`, if I did `X-GitHubEvent` (which is how the docs refer to the header) TS will give me a error alerting me to the typo (which is accurate as Azure downcases the headers).

Reply:
Ohhh sure - great point!! 

@alrod - could you change back to `[name: string]: string | undefined`? In these cases it's always known to only be a string value or undefined (nothing more, nothing less)